src/bots/fillerLite.ts

```diff
@@ -0,0 +1,1274 @@
+
+					if (isVariant(actionRecord.action, 'fill')) {
+						if (isVariant(actionRecord.marketType, 'perp')) {
+							const perpMarket = this.driftClient.getPerpMarketAccount(
+								actionRecord.marketIndex
+							);
+							if (perpMarket) {
+								this.observedFillsCountCounter.add(1, {
+									market: decodeName(perpMarket.name),
+								});
+							}
+						}
+					}
+				}
+			}
+		);
+
+		logger.info(`${this.name} Bot started! (websocket: true)`);
+	}
+
+	public async reset() {
+		for (const intervalId of this.intervalIds) {
+			clearInterval(intervalId);
+		}
+		this.intervalIds = [];
+
+		this.eventSubscriber.eventEmitter.removeAllListeners('newEvent');
+
+		await this.orderSubscriber.unsubscribe();
+		await this.userStatsMap.unsubscribe();
+	}
+
+	protected async getNodeToFillInfo(nodeToFill: NodeToFill): Promise<{
+		makerInfos: Array<MakerInfo> | undefined;
+		takerUser: UserAccount;
+		referrerInfo: ReferrerInfo;
+		marketType: MarketType;
+	}> {
+		const makerInfos: Array<MakerInfo> | undefined = [];
+
+		// set to track whether maker account has already been included
+		const makersIncluded = new Set<string>();
+		if (nodeToFill.makerNodes.length > 0) {
+			for (const makerNode of nodeToFill.makerNodes) {
+				if (this.isDLOBNodeThrottled(makerNode)) {
+					continue;
+				}
+
+				const makerAccount = makerNode.userAccount.toBase58();
+				if (makersIncluded.has(makerAccount)) {
+					continue;
+				}
+
+				// Is must get still necessary here? NodeToFill is coming from orderSubscriber,
+				// which also caches the userAccounts in a Map. So get SHOULD be sufficient.
+				const makerUserAccount =
+					this.orderSubscriber.usersAccounts.get(makerAccount).userAccount;
+				const makerAuthority = makerUserAccount.authority;
+
+				const makerUserStats = (
+					await this.userStatsMap.mustGet(makerAuthority.toString())
+				).userStatsAccountPublicKey;
+				makerInfos.push({
+					maker: makerNode.userAccount,
+					makerUserAccount: makerUserAccount,
+					order: makerNode.order,
+					makerStats: makerUserStats,
+				});
+				makersIncluded.add(makerAccount);
+			}
+		}
+
+		const takerUser = this.orderSubscriber.usersAccounts.get(
+			nodeToFill.node.userAccount.toString()
+		);
+
+		const referrerInfo = (
+			await this.userStatsMap.mustGet(
+				takerUser.userAccount.authority.toString()
+			)
+		).getReferrerInfo();
+
+		return Promise.resolve({
+			makerInfos,
+			takerUser: takerUser.userAccount,
+			referrerInfo,
+			marketType: nodeToFill.node.order.marketType,
+		});
+	}
+
+	/**
+	 * Iterates through a tx's logs and handles it appropriately 3e.g. throttling users, updating metrics, etc.)
+	 *
+	 * @param nodesFilled nodes that we sent a transaction to fill
+	 * @param logs logs from tx.meta.logMessages or this.clearingHouse.program._events._eventParser.parseLogs
+	 *
+	 * @returns number of nodes successfully filled
+	 */
+	protected async handleTransactionLogs(
```

Comment:
a lot of this is same as the parent implementation, perhaps pass in a function to retrieve the `UserAccount`, then filler lite can use `orderSubscriber`, and parent can keep using `UserMap`?